Improve README (#25)
ostis-ai/ostis-web-platform: README.md

# OSTIS Web Platform

> <img src="https://github.com/ostis-ai/ostis-web-platform/actions/workflows/main.yml/badge.svg?branch=develop">

Reviewer: Can we make own badge? (maybe, OSTIS or SC-code, or LaTeX, or another one)

Reviewer: We can also add license badge

Author: [shields.io](https://shields.io/) is a great website which has tons of badges. You can check out the possible options here, though I don't think we need a badge of our own for now. A license badge is a good idea though, I'll add it

> <img src="https://github.com/ostis-ai/ostis-web-platform/actions/workflows/main.yml/badge.svg?branch=develop">

Reviewer 2: Add License icon

This repository is a web-oriented software platform of [OSTIS Project](https://github.com/ostis-ai/ostis-project), and it is intended to be a solid framework to help you deploy existing and create new OSTIS systems.

OSTIS Web platform contains:

1) [Knowledge base with top-level ontologies](https://github.com/ostis-ai/ims.ostis.kb) to help you develop a broad variety of information models

> 2) [Knowledge processing machine](https://github.com/ostis-ai/sc-machine) with semantic network storage and agent-based knowledge processing

Reviewer 2: sc-machine is not KPM, we need to find better formulation

3) [Web-oriented semantic interface](https://github.com/ostis-ai/sc-web) for users to interact with the intelligent system.

> To learn more about the platform, check out our [documentation](https://raw.githubusercontent.com/ostis-ai/ostis-web-platform/develop/docs/main.pdf).

Reviewer 2: this link push me to download pdf file. I think, better to use https://github.com/ostis-ai/ostis-web-platform/blob/develop/docs/main.pdf

## Installation

+ Quick start using Docker Compose

   Requirements: you will need [Docker](https://docs.docker.com/get-docker/) installed and running on your machine.

   ```sh
   git clone https://github.com/ostis-ai/ostis-web-platform
   cd ostis-web-platform
   # download images from Docker Hub
   docker compose pull
   # build knowledge base
   docker compose run machine build
   # launch web platform stack
   docker compose up
   ```

   <details>
   <summary> Building docker images locally </summary>

>    This may come in handy e.g. when you want to use a custom branch of sc-machine or sc-web.

Reviewer 3: ```suggestion
   This may come in handy e.g. when you want to use a custom branch of the sc-machine or sc-web.
```

   ### Requirements

   1. **If you're using Windows**, please make sure you are using UNIX line endings inside the repository and `longpaths` are enabled:

      ```
      git config --local core.autocrlf true
      git config --local core.longpaths true
      git add --normalize .

>       ```

Reviewer 4: As I know, we should face these problems on Windows even without building (e.g. cloning `ims.ostis.kb`). So maybe it'd be better not to hide these lines in `details` section.

Author: This is definitely a valid concern, but we should probably test it. Also I've seen a weird behaviour on a Windows machine - it required the `git add` flag to be `--renormalize`, not `--normalize`. Long story short - testing required before merging this, and, well, I don't have a Windows machine. It'd be great to get some help with that 😄

Author: Soo... What are we doing about it? Did we test that **not** having these settings actually breaks the KB building process on Windows?

   2. Enable Docker BuildKit. You can use `DOCKER_BUILDKIT=1` shell variable for this.

   ### Build process

   ```sh
   git clone https://github.com/ostis-ai/ostis-web-platform
   git submodule update --init --recursive
   cd scripts
   ./prepare.sh no_build_sc_machine no_build_sc_web # download all submodules without compilation.
   cd ..
   docker compose build
   ```

   </details>

+ Natively

>    Note: Currently only Ubuntu is supported by this installation method. If you're going to use it, it might take a while to download dependencies and compile the components. Use it only if you know what you're doing!

Reviewer 3: ```suggestion
   Note: Currently, only Ubuntu is supported by this installation method. If you're going to use it, it might take a while to download dependencies and compile the components. Use it only if you know what you're doing!
```

   ```sh
   cd ostis-web-platform/scripts/
   ./prepare.sh
   ```

## Usage

+ Docker Compose

   ```sh
   # build the Knowledge Base.
   # Required before the first startup (or if you've made updates to KB sources)
   docker compose run machine build
   # start platform services and run web interface at localhost:8000
   docker compose up
   ```

+ Native installation

   ```sh
   # Build knowledge base
   cd ostis-web-platform/scripts/
   ./build_kb.sh
   # Launch knowledge processing machine
   cd ostis-web-platform/scripts/
   ./run_sc_server.sh
   # *in another terminal*
   # Launch semantic web interface at localhost:8000
   cd ostis-web-platform/scripts/
   ./run_scweb.sh
   ```

## Documentation

> We document all information about project development and its components' implementation in sources of its knowledge base,

Reviewer 3: ```suggestion
We document all information about the project development and its components' implementation in sources of its knowledge base
```

to provide opportunity to use it in information processing and knowledge generation.

You can access the current version of the documentation in [docs/main.pdf](docs/main.pdf) file of this project.

Documentation is written with

> the help of LaTeX tools in SCn-code representation. To build documentation manually, you'll need a LaTeX distribution installed on your computer. Alternatively we're providing a Docker image to build the documentation in case you can't / don't want to install LaTeX on your PC.

Reviewer 3: ```suggestion
the help of LaTeX tools in SCn-code representation. To build documentation manually, you'll need a LaTeX distribution installed on your computer. Alternatively, we provide a Docker image to build the documentation in case you can't / don't want to install LaTeX on your PC.
```

### Download subprojects documentation

```sh
# feel free to skip this step if the platform is already installed natively
cd ostis-web-platform/scripts
./prepare.sh no_build_sc_machine no_build_sc_web
```

+ ### Build steps (using LaTeX)

   ```sh
   cd ostis-web-platform/docs
   latexmk -pdf -bibtex main.tex
   ```

+ ### Build steps (using Docker)

   ```sh
   cd ostis-web-platform
   docker run -v ${PWD}:/workdir -w /workdir/docs --rm -it ostis/scn-latex-plugin:latest "main.tex"
   ```

   After the compilation, the `main.pdf` file should appear at `ostis-web-platform/docs/`.

## Feedback

> Contributions, bug reports and feature requests are welcome! Feel free to check our [issues page](https://github.com/ostis-ai/ostis-web-platform/issues) and file a new issue (or comment in existing ones).

Reviewer 3: ```suggestion
Contributions, bug reports, and feature requests are welcome! Feel free to check our [issues page](https://github.com/ostis-ai/ostis-web-platform/issues) and file a new issue (or comment in existing ones).
```

Author: Oxford comma is a bit disorienting for a Russian-speaking auditory, so I think it's better to skip it.
